diff --git a/protoc-gen-grpc-gateway/internal/gengateway/template.go b/protoc-gen-grpc-gateway/internal/gengateway/template.go
index 6392af9..2d28493 100644
--- a/protoc-gen-grpc-gateway/internal/gengateway/template.go
+++ b/protoc-gen-grpc-gateway/internal/gengateway/template.go
@@ -152,13 +152,13 @@ type trailerParams struct {
 func typeFromName(name string) string {
 	lowerName := strings.ToLower(name)
 	if strings.Contains(lowerName, "epoch") {
-		return "github_com_prysmaticlabs_prysm_v4_consensus_types_primitives.Epoch"
+		return "github_com_prysmaticlabs_prysm_v5_consensus_types_primitives.Epoch"
 	} else if strings.Contains(lowerName, "slot") {
-		return "github_com_prysmaticlabs_prysm_v4_consensus_types_primitives.Slot"
+		return "github_com_prysmaticlabs_prysm_v5_consensus_types_primitives.Slot"
 	} else if strings.Contains(lowerName, "committee") {
-		return "github_com_prysmaticlabs_prysm_v4_consensus_types_primitives.CommitteeIndex"
+		return "github_com_prysmaticlabs_prysm_v5_consensus_types_primitives.CommitteeIndex"
 	} else if strings.Contains(lowerName, "index") {
-		return "github_com_prysmaticlabs_prysm_v4_consensus_types_primitives.ValidatorIndex"
+		return "github_com_prysmaticlabs_prysm_v5_consensus_types_primitives.ValidatorIndex"
 	}
 	return ""
 }
@@ -242,7 +242,7 @@ It translates gRPC into RESTful JSON APIs.
 */{{end}}
 package {{.GoPkg.Name}}
 import (
-	github_com_prysmaticlabs_prysm_v4_consensus_types_primitives "github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	github_com_prysmaticlabs_prysm_v5_consensus_types_primitives "github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
 	"google.golang.org/protobuf/types/known/emptypb"
 	{{range $i := .Imports}}{{if $i | printf "%q" | ne "github.com/golang/protobuf/ptypes/empty"}}{{$i | printf "%s\n"}}{{end}}{{end}}
 )
@@ -254,7 +254,7 @@ var _ status.Status
 var _ = runtime.String
 var _ = utilities.NewDoubleArray
 var _ = metadata.Join
-var _ = github_com_prysmaticlabs_prysm_v4_consensus_types_primitives.Epoch(0)
+var _ = github_com_prysmaticlabs_prysm_v5_consensus_types_primitives.Epoch(0)
 var _ = emptypb.Empty{}
 `))
 
