diff --git a/kafka/BUILD.bazel b/kafka/BUILD.bazel
index bc46110..367c9f6 100644
--- a/kafka/BUILD.bazel
+++ b/kafka/BUILD.bazel
@@ -29,19 +29,20 @@ go_library(
         "testhelpers.go",
         "time.go",
     ],
+    cdeps = ["//kafka/librdkafka:precompiled"],
     cgo = True,
     clinkopts = select({
         "@io_bazel_rules_go//go/platform:android": [
-            "kafka/librdkafka/librdkafka_glibc_linux.a -lm -ldl -lpthread -lrt",
+            "-lm -ldl -lpthread -lrt",
         ],
         "@io_bazel_rules_go//go/platform:darwin": [
-            "kafka/librdkafka/librdkafka_darwin.a -lm -lsasl2 -lz -ldl -lpthread",
+            "-lm -lsasl2 -lz -ldl -lpthread",
         ],
         "@io_bazel_rules_go//go/platform:ios": [
-            "kafka/librdkafka/librdkafka_darwin.a -lm -lsasl2 -lz -ldl -lpthread",
+            "-lm -lsasl2 -lz -ldl -lpthread",
         ],
         "@io_bazel_rules_go//go/platform:linux": [
-            "kafka/librdkafka/librdkafka_glibc_linux.a -lm -ldl -lpthread -lrt",
+            "-lm -ldl -lpthread -lrt",
         ],
         "//conditions:default": [],
     }),
diff --git a/kafka/librdkafka/BUILD.bazel b/kafka/librdkafka/BUILD.bazel
index 2ced242..9c06d83 100644
--- a/kafka/librdkafka/BUILD.bazel
+++ b/kafka/librdkafka/BUILD.bazel
@@ -8,4 +8,26 @@ go_library(
     ],
     importpath = "gopkg.in/confluentinc/confluent-kafka-go.v1/kafka/librdkafka",
     visibility = ["//visibility:public"],
+    cgo = True,
+    cdeps = [":precompiled"],
+)
+
+cc_library(
+    name = "precompiled",
+    srcs = select({
+	"@io_bazel_rules_go//go/platform:android": [ 
+		"librdkafka_glibc_linux.a",
+	],
+	"@io_bazel_rules_go//go/platform:darwin": [
+		"librdkafka_darwin.a",
+	],
+	"@io_bazel_rules_go//go/platform:ios": [
+		"librdkafka_darwin.a",
+	],
+	"@io_bazel_rules_go//go/platform:linux": [
+		"librdkafka_glibc_linux.a",
+	],
+   }),
+    visibility = ["//visibility:public"],
+    hdrs = ["rdkafka.h"],
 )
