diff --git a/crypto.go b/crypto.go
index e6d6d5f..f9eb389 100644
--- a/crypto.go
+++ b/crypto.go
@@ -217,10 +217,7 @@ func preferServerCipherSuites() bool {
 	var (
 		hasGCMAsmAMD64 = cpu.X86.HasAES && cpu.X86.HasPCLMULQDQ
 		hasGCMAsmARM64 = cpu.ARM64.HasAES && cpu.ARM64.HasPMULL
-		// Keep in sync with crypto/aes/cipher_s390x.go.
-		hasGCMAsmS390X = cpu.S390X.HasAES && cpu.S390X.HasAESCBC && cpu.S390X.HasAESCTR && (cpu.S390X.HasGHASH || cpu.S390X.HasAESGCM)
-
-		hasGCMAsm = hasGCMAsmAMD64 || hasGCMAsmARM64 || hasGCMAsmS390X
+		hasGCMAsm      = hasGCMAsmAMD64 || hasGCMAsmARM64
 	)
 	return !hasGCMAsm
 }
